src/hotspot/os/windows/os_windows.cpp

```diff
@@ -1279,8 +1279,33 @@ void  os::dll_unload(void *lib) {
   }
 }
 
+static size_t format_message_fixup(char* buf, size_t end) {
+  if (end > 3) {
+    if (buf[end - 1] == '\n') end--;
+    if (buf[end - 1] == '\r') end--;
+    if (buf[end - 1] == '.')  end--;
+    buf[end] = '\0';
+  }
+  return end;
```

Comment:
This should have copied the explanatory comment as well. 

Pre-existing, but is  "./r/n" always added? If so the 'if' checks look weird - why would you not just set `buf[end-3] = '\0'` directly? If the formatting is actually variable this code will remove ".", "/r", "/n", "./r", "./n", "/r/n" and "./r/n". ???

Reply:
Ok.

I have no clue.